nin/dasBoot/NodeManager.js

```diff
@@ -0,0 +1,132 @@
+class NodeManager {
+
+  constructor() {
+    this.nodes = {};
+    this.graphChangeListeners = [];
+  }
+
+  onGraphChange(fn) {
+    this.graphChangeListeners.push(fn);
+  }
+
+  fireGraphChange() {
+    console.log('graph change!');
+    for(var i = 0; i < this.graphChangeListeners.length; i++) {
+      this.graphChangeListeners[i]();
+    }
+  }
+
+  createNode(nodeInfo) {
+    nodeInfo.options = nodeInfo.options || {};
+    var node;
+    if(nodeInfo.type.slice(0, 4) == 'NIN.') {
+      node = new NIN[nodeInfo.type.slice(4)](nodeInfo.id, nodeInfo.options);
+    } else {
+      node = new window[nodeInfo.type](nodeInfo.id, nodeInfo.options);
+    }
+    for(var key in node.inputs) {
+      node.inputs[key].node = node;
+    }
+    for(var key in node.outputs) {
+      node.outputs[key].node = node;
+    }
+    return node;
+  }
+
+  insertNode(node) {
```

Comment:
ReplaceNode might be more accurate?
I when i encountered it further down (in the WS node push reload code) i got confused by the insertion.